### PR TITLE
Some followups to #216

### PR DESCRIPTION
**Pull request by nrc**
The first commit separate manifest package into two to keep code for old (v0) and new (v1) manifests distinct. The second commit removes `description` from `ComponentItem`. The description is kept only in the component manifest, not the index so that the field is not duplicated and so that an owner can update a component including its description without changing the index.

PTAL @lonng

### CODE REVIEW

File: cmd/run.go [hunk 1]
```diff
@@ -17,6 +17,7 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
+	"github.com/pingcap-incubator/tiup/pkg/repository/v0manifest"
```
Comment (lonng) [inline cmd/run.go:20]: Please optimize all imports.

File: tools/migrate/main.go [hunk 2]
```diff
@@ -254,22 +255,21 @@ func migrate(srcDir, dstDir string) error {
 			return errors.Trace(err)
 		}
 
-		index.Components[comp.Name] = manifest.ComponentItem{
-			Name:        comp.Name,
-			Description: comp.Desc,
```
Comment (lonng) [inline tools/migrate/main.go:259]: The description is used at `tiup list`, we must fetch all component manifest to display the description if we remove it from `ComponentItem`.

Comment (nrc) [reply tools/migrate/main.go:259]: I think it is ok - we will have all the manifests locally and we can cache the descriptions locally too.

File: cmd/run.go (hunk 1 above)
Comment (nrc) [reply cmd/run.go:20]: is there an argument to goimports which will make it do this?

Comment (lonng) [reply cmd/run.go:20]: I think `make fmt` will optimize the imports.

Comment (nrc) [reply cmd/run.go]: It only organises them within groups :-(

**Comment by nrc** [comment]
@lonng please take another look

**Pull request merged**